sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -754,57 +754,69 @@ static bool loadDeviceLib(const ContextImplPtr Context, const char *Name,
   return Prog != nullptr;
 }
 
-static const char *getDeviceLibFilename(DeviceLibExt Extension) {
-  switch (Extension) {
-  case DeviceLibExt::cl_intel_devicelib_assert:
-    return "libsycl-fallback-cassert.spv";
-  case DeviceLibExt::cl_intel_devicelib_math:
-    return "libsycl-fallback-cmath.spv";
-  case DeviceLibExt::cl_intel_devicelib_math_fp64:
-    return "libsycl-fallback-cmath-fp64.spv";
-  case DeviceLibExt::cl_intel_devicelib_complex:
-    return "libsycl-fallback-complex.spv";
-  case DeviceLibExt::cl_intel_devicelib_complex_fp64:
-    return "libsycl-fallback-complex-fp64.spv";
-  case DeviceLibExt::cl_intel_devicelib_cstring:
-    return "libsycl-fallback-cstring.spv";
-  case DeviceLibExt::cl_intel_devicelib_imf:
-    return "libsycl-fallback-imf.spv";
-  case DeviceLibExt::cl_intel_devicelib_imf_fp64:
-    return "libsycl-fallback-imf-fp64.spv";
-  }
-  throw compile_program_error("Unhandled (new?) device library extension",
-                              PI_ERROR_INVALID_OPERATION);
+static const std::map<DeviceLibExt,
+                      std::pair<const std::string, const std::string>>
```

Comment:
Is there a reason to save them as strings or could we go back to using `const char *`?

Reply:
const char* is simpler. Made the change.